drivers/i2c/target/eeprom_target.c

```diff
@@ -50,6 +50,13 @@ unsigned int eeprom_target_get_size(const struct device *dev)
 	return data->buffer_size;
 }
 
+uint8_t *eeprom_target_get_data(const struct device *dev)
+{
+	struct i2c_eeprom_target_data *data = dev->data;
+
+	return data->buffer;
```

Comment:
This is typically frowned upon as it breaks encapsulation. You are exposing driver internal data structures here. How about adding proper read/write APIs (similar to those in `include/zephyr/drivers/eeprom.h`) and deprecate the old APIs?

Reply:
That is also fine by me. How exactly do I go about deprecating the old API? Any suggestion for the function names given that the legacy API already has a `eeprom_target_read()`?